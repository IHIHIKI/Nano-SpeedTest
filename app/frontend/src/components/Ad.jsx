import React from 'react';
import '../styles/Ad.css';

export default () => {
    return (
        <div className='container'>
            <div className='jumbotron'>
                <div id='ad-content'>
                    <a href='https://nano21.net' className='row'>
                        <b>â™  Nano21.net </b>&nbsp; | 20 Nano Give-Away for testers! | Nano Blackjack Casino
                    </a>
                    <p className='row'>COMMUNITY AD</p>
                </div>
            </div>
        </div>
    );
};

