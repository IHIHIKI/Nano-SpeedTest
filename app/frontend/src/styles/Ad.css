.container {
    margin-top: 10px;
}
.jumbotron {
    padding: 15px 30px;
    margin-bottom: 0;
    background-color: dodgerblue;
}
.jumbotron > p {
    text-align: left;
    margin: 0;
}
p > a {
    color: #f3f3f3;
}
p > p {
    color: gainsboro;
    font-size: 70%;
    font-weight: 600;
    text-decoration: underline;
    margin: 0 5px;
}

p > a:hover{
    color: gainsboro;
    text-decoration: none;
}

p > a:hover{
    color: gainsboro;
    text-decoration: none;
}