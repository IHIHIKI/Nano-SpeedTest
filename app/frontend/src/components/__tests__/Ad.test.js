import React from 'react';
import renderer from 'react-test-renderer';
import Ad from '../Ad';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { BrowserRouter} from 'react-router-dom';
import { shallow } from 'enzyme';

const mockStore = configureStore();
let store = mockStore({
    ads: {
        currentAd: {
            title: 'Title',
            message: 'message',
            url: 'www.test.com'
        }
    }
});

it('shallow renders without crashing', () => {
    shallow(<Ad store={store}/>);
});

it(`renders snapshot correctly`, () => {
    const tree = renderer.create(
        <Provider store={store}>
            <BrowserRouter >
                <Ad />
            </BrowserRouter>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});