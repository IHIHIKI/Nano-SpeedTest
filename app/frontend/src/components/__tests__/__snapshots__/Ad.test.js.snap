// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot correctly 1`] = `
<div
  className="container"
>
  <div
    className="jumbotron"
  >
    <div
      id="ad-content"
    >
      <a
        className="row"
        href="www.test.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        <b>
          Title
        </b>
        Â 
        message
      </a>
      <p
        className="row"
      >
        COMMUNITY AD
      </p>
    </div>
  </div>
</div>
`;
